qiskit/backends/local/localprovider.py

```diff
@@ -52,7 +52,7 @@ def available_backends(self):
         # pylint: disable=arguments-differ
         return list(self.backends.values())
 
-    def aliased_backend_names(self):
+    def grouped_backend_names(self):
```

Comment:
All this renaming from `aliases` to `grouped` is a breaking change since these methods are public but I think they should be private. Actually, this seems to return a _static_ dictionary. If we don't mind about breaking something in the 3rd party code, I would recommend converting `aliased_backend_names` into a private property `_backend_aliases`.

Also, I think _alias_ is a more accurate name. _Group_/_grouped_ seems to convey some unity meaning which is not the case: there is no group called `local_qasm_simulator`, it simply turns out that `local_qasm_simulator_cpp`, `local_qasm_simulator_py` and `local_clifford_simulator_cpp` are other forms of referring `local_qasm_simulator`.

Reply:
Yes, these should be private. I don't mind breaking, since I am sure almost no 3rd party code uses them.
I don't mind method vs. property.

Group is more accurate. The simulators are grouped by functionality. They may have different implementations, but fundamentally they do the same thing.

I am using alias to just refer to a mere renaming, e.g. ibmqx4 -> ibmq_5_tenerife

Reply:
I see, it does not seem clear to me but I'm fine. Leave it this way.